### PR TITLE
Update RFD 12 with image tagging scheme

### PR DESCRIPTION
**Pull request by awly**
We don't have to do it right away, it describes the ideal final state.

@webvictim @benarent PTAL, wanted to document our discussion.
@webvictim, I understand it'll take extra work in the drone pipeline to deal with all these versions. We can stick with what we have for now and add `teleport:X`-like tags with v6.

Updates https://github.com/gravitational/teleport/issues/5216

### CODE REVIEW

**Review by webvictim — COMMENTED** [review]
This looks fine to me. I think the main point is that if we're going to stick with `teleport:x.y` tags once we switch to `teleport:x` then we will still need to build a dynamic number of `x.y` tags for every intermediate release, which is complicated and undesirable. I don't see why we'd want to keep offering `x.y` if `x` is guaranteed to be backward compatible.

**Comment by awly** [comment]
I don't think we _want_ to keep offering `x.y`. It's just to avoid questions:

> In practice, there should be no reason to use `teleport:X.Y` tags over
`teleport:X` due to our compatibility guarantees. `teleport:X.Y` tags exist
only to avoid user confusion and questions.

I guess I don't have the right mental model of how our docker images are created.
When we cut `X.Y.Z` and all the drone release machinery runs, it seems trivial to expand `X.Y.Z` to `[X.Y.Z, X.Y, X]` and push all those tags.

**Comment by webvictim** [comment]
The nightly releases (currently `teleport:X.Y`, proposed to be `teleport:X` as of 6.0) are [built by a cron job](https://github.com/gravitational/teleport/blob/master/.drone.yml#L474-L614) which is a completely separate set of logic from the [tagged Docker image builds](https://github.com/gravitational/teleport/blob/master/.drone.yml#L2536-L2643) and [the pushes which happen when we promote a release](https://github.com/gravitational/teleport/blob/master/.drone.yml#L3016-L3053).

I think the best solution here is to make an announcement to customers that we're moving to `teleport:X` cron images rather than `teleport:X.Y` and update the cron job logic to only build the `teleport:X` tag, plus any supported older releases. Continuing to offer `teleport:X.Y` images when `X.Y` and `X.Y-2` are already guaranteed to be compatible seems redundant and confusing - but I do think we should continue to do it until the 6.0 release because we haven't communicated the change properly yet.

File: rfd/0012-teleport-versioning.md [hunk 1]
```diff
@@ -141,3 +141,32 @@ story. We never want a `vN+1` that has no migration path other than "rebuild
...
+  `4.4.1`, `4.4.2`)
+- `teleport:X.Y` - mutable tag pointing at the latest patch release for `X.Y` (for
+  example `4.4` pointing at `4.4.2`)
+
+For new versioning, the following tags are created:
+
+- `teleport:X.Y.Z` - immutable tag for a specific patch release (for example
+  `5.1.0`, `5.1.1`)
+- `teleport:X.Y` - mutable tag pointing at the latest patch release for minor
+  version `X.Y` (for example `5.1` pointing at `5.1.1`)
```
Comment (webvictim) [inline rfd/0012-teleport-versioning.md:159]: To be clear, I think we should just stop offering these to encourage people to switch to the new `teleport:X` format.

**Comment by awly** [comment]
@webvictim removed `teleport:X.Y` tag.
Let's go with your suggestion and trade some initial customer confusion for lower long-term maintenance burden.

**Review by webvictim — APPROVED** [review]

**Comment by awly** [comment]
Still need approval from @benarent or @klizhentas or @russjones

**Comment by awly** [comment]
Ping @benarent @klizhentas @russjones

**Review by klizhentas — APPROVED** [review]

**Pull request merged**